packages/flutter/test/gestures/monodrag_test.dart

```diff
@@ -10,6 +10,31 @@ import 'gesture_tester.dart';
 void main() {
   TestWidgetsFlutterBinding.ensureInitialized();
 
+  test('acceptGesture tolerates a null lastPendingEventTimestamp', () {
+    // Regression test for https://github.com/flutter/flutter/issues/112403
+    // and b/249091367
+    final DragGestureRecognizer recognizer = VerticalDragGestureRecognizer();
+    const PointerDownEvent event = PointerDownEvent(timeStamp: Duration(days: 10));
+
+    expect(recognizer.debugLastPendingEventTimestamp, null);
+
+    recognizer.addAllowedPointer(event);
+    expect(recognizer.debugLastPendingEventTimestamp, event.timeStamp);
+
+    // Normal case: acceptGesture called and we have a last timestamp set.
+    recognizer.acceptGesture(event.pointer);
+    expect(recognizer.debugLastPendingEventTimestamp, null);
+
+    // Reject the gesture to reset state and allow accepting it again.
+    recognizer.rejectGesture(event.pointer);
+    expect(recognizer.debugLastPendingEventTimestamp, null);
+
+    // Not entirely clear how this can happen, but the bugs mentioned above show
+    // we can end up in this state empircally.
```

Comment:
This would really be interesting to figure out... Wondering if that's a bug somewhere in the framework in how we use the gesture detector, or in user code, or do the embedders send bogus events??

(Not actionable for this PR)

Reply:
Yes, I suspect there's something off here but I don't know that subsystem well enough to know. I'm half hoping one of the people I tagged on this PR will know :)